test/elixir/test/mango/05_index_selection_test.exs

```diff
@@ -0,0 +1,457 @@
+    # due to an explicit exists clause
+    selector = %{"company" => "Pharmex", "manager" => %{"$exists" => false}}
+    {:ok, docs} = MangoDatabase.find(@db_name, selector)
+    assert length(docs) == 1
+    assert Enum.at(docs, 0)["company"] == "Pharmex"
+    assert "manager" != Enum.at(docs, 0)
+
+    {:ok, resp_explain} = MangoDatabase.find(@db_name, selector, explain: true)
+    assert resp_explain["index"]["type"] == "special"
+  end
+end
+
+defmodule TextIndexSelectionTest do
+  use CouchTestCase
+
+  @db_name "text-index-selection"
+
+  setup do
+    if MangoDatabase.has_text_service() do
+      UserDocs.setup(@db_name, "text")
+    else
+      {:skip, "requires text service"}
+    end
+  end
+
+  test "with text" do
+    {:ok, resp} = MangoDatabase.find(@db_name,
+      %{
+        "$text" => "Stephanie",
+        "name.first" => "Stephanie",
+        "name.last" => "This doesn't have to match anything.",
+      },
+      explain: true,
+    )
+    assert resp["index"]["type"] == "text"
+  end
+
+  test "no view index" do
+    {:ok, resp} = MangoDatabase.find(@db_name, %{"name.first" => "Ohai!"}, explain: true)
+    assert resp["index"]["type"] == "text"
+  end
+
+  test "with or" do
+    {:ok, resp} = MangoDatabase.find(@db_name,
+    %{
+      "$or": [
+          %{"name.first" => "Stephanie"},
+          %{"name.last" => "This doesn't have to match anything."},
+        ]
+      },
+      explain: true,
+    )
+    assert resp["index"]["type"] == "text"
+  end
+
+  test "manual bad text idx" do
+    design_doc = %{
+      "_id" => "_design/bad_text_index",
+      "language" => "query",
+      "indexes" => %{
+        "text_index" => %{
+          "default_analyzer" => "keyword",
+          "default_field" => %{},
+          "selector" => %{},
+          "fields" => "all_fields",
+          "analyzer" => %{
+            "name" => "perfield",
+            "default" => "keyword",
+            "fields" => %{"$default" => "standard"},
+          },
+        }
+      },
+      "indexes" => %{
+        "st_index" => %{
+          "analyzer" => "standard",
+          "index" => 'function(doc){\n index("st_index", doc.geometry);\n}',
+        }
+      },
+    }
+    MangoDatabase.save_docs(@db_name, design_doc)
+    {:ok, docs} = MangoDatabase.find(@db_name, %{"age" => 48})
+    assert length(docs) == 1
+    assert Enum.at(docs, 0)["name"]["first"] == "Stephanie"
+    assert Enum.at(docs, 0)["age"] == 48
+  end
+end
+
+defmodule MultiTextIndexSelectionTest do
+  use CouchTestCase
+
+  @db_name "multi-text-index-selection"
+
+  setup do
+    if MangoDatabase.has_text_service() do
+      UserDocs.setup(@db_name, "text")
+      MangoDatabase.create_text_index(@db_name, ddoc: "foo", analyzer: "keyword")
+      MangoDatabase.create_text_index(@db_name, ddoc: "bar", analyzer: "email")
+      :ok
+    else
+      {:skip, "requires text service"}
+    end
+  end
+
+  test "fallback to json with multi text" do
```

Comment:
this one is throwing error
`"You must specify an index with the use_index parameter."`
not sure what i'm missing 🤷